source/common/network/connection_impl.cc

```diff
@@ -535,6 +586,34 @@ bool ConnectionImpl::bothSidesHalfClosed() {
   return read_end_stream_ && write_end_stream_ && write_buffer_->length() == 0;
 }
 
+#ifdef __APPLE__
+bool ConnectionImpl::detectEarlyClose() {
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+
+  if (is_uds_) {
```

Comment:
does uds support this on non-apple?  I thought we already discovered that was broken?

Reply:
Fair. It can just return false for this case.